storage/client.go

```diff
@@ -179,6 +179,44 @@ func (e UnexpectedStatusCodeError) Got() int {
 	return e.got
 }
 
+// NewClientFromConnectionString creates a Client from the connection string.
+func NewClientFromConnectionString(input string) (Client, error) {
+	var (
+		accountName, accountKey, endpointSuffix string
+		useHTTPS                                = defaultUseHTTPS
+	)
+
+	for _, pair := range strings.Split(input, ";") {
+		if pair == "" {
+			continue
+		}
+
+		equalDex := strings.IndexByte(pair, '=')
+		if equalDex <= 0 {
+			return Client{}, fmt.Errorf("Invalid connection segment %q", pair)
```

Comment:
Nit: we know this is a string, so I'd rather have the output format surround it in quotes and have the format specifier `%s`.

Reply:
`%q` does format the parameter correct:

> %q	a double-quoted string safely escaped with Go syntax

For this reason I generally prefer to use it over manually quoting a %s; there's no confusion if the string happens to also contain quotes.

Reply:
Ah, beautiful. Thanks.

Reply:
For people curious: https://play.golang.org/p/lVBlTRB2OH